.circleci/config.yml

```diff
@@ -373,6 +373,42 @@ jobs:
           key: version-cache-{{ checksum "old_versions.txt" }}
           paths:
             - /home/circleci/experimenter/old_versions.txt
+  build_rust_image:
+    working_directory: ~/experimenter
+    machine:
+      image: ubuntu-2004:202107-02 # Ubuntu 20.04, Docker v20.10.7, Docker Compose v1.29.2
+      docker_layer_caching: true
+    steps:
+      - checkout
+      - run:
+          name: Check for tag change
+          command: |
+            set +e
+            docker pull ${DOCKERHUB_REPO}:nimbus-rust-image
+            docker run -d --name firefox-beta ${DOCKERHUB_REPO}:nimbus-rust-image
+            docker_id=$(docker ps -aqf "name=^nimbus-rust-image")
+            docker cp $docker_id:/application-services-old.txt /home/circleci/experimenter/application-services-old.txt
+            git -c 'versionsort.suffix=-' \
+              ls-remote --exit-code --refs --sort='version:refname' --tags https://github.com/mozilla/application-services.git \
+              '*.*.*' | tail --lines=1 | cut --delimiter='/' --fields=3 > application-services-current.txt
+            DIFF=$(diff /home/circleci/experimenter/application-services-current.txt /home/circleci/experimenter/application-services-old.txt)
+            if [ ! "$DIFF" ]; then
+                echo "No AS updates"
+                circleci-agent step halt
+            fi
+      - run: 
+          name: Build rust test image
+          command: |
+            set +e
+            AS_VERSION=$(git -c 'versionsort.suffix=-' \
+              ls-remote --exit-code --refs --sort='version:refname' --tags https://github.com/mozilla/application-services.git \
+              '*.*.*' | tail --lines=1 | cut --delimiter='/' --fields=3)
+            docker build -t ${DOCKERHUB_REPO}:nimbus-rust-image -f app/tests/integration/nimbus/utils/Dockerfile-rust-image --build-arg as_version=$AS_VERSION --progress=plain .
```

Comment:
Ah so this gets the ref for the latest version based on the git tags and uses it as the a-s version build arg?

Reply:
Yes, kudos to stack overflow 😄